pythonforandroid/graph.py

```diff
@@ -99,17 +132,129 @@ def find_order(graph):
                 bset.discard(result)
 
 
-def get_recipe_order_and_bootstrap(ctx, names, bs=None):
-    recipes_to_load = set(names)
+def obvious_conflict_checker(ctx, name_tuples, blacklist=[]):
+    """ This is a pre-flight check function that will completely ignore
+        recipe order or choosing an actual value in any of the multiple
+        choice tuples/dependencies, and just do a very basic obvious
+        conflict check.
+    """
+    deps_were_added_by = dict()
+    deps = set()
+
+    # Add dependencies for all recipes:
+    to_be_added = [(name_tuple, None) for name_tuple in name_tuples]
+    while len(to_be_added) > 0:
+        current_to_be_added = list(to_be_added)
+        to_be_added = []
+        for (added_tuple, adding_recipe) in current_to_be_added:
+            assert(type(added_tuple) == tuple)
+            if len(added_tuple) > 1:
+                # No obvious commitment in what to add, ignore it.
+                # (Remember this function only catches obvious issues)
+                continue
+
+            name = added_tuple[0]
+            recipe_conflicts = set()
+            recipe_dependencies = []
+            try:
+                # Get recipe to add and who's ultimately adding it:
+                recipe = Recipe.get_recipe(name, ctx)
+                recipe_conflicts = {c.lower() for c in recipe.conflicts}
+                recipe_dependencies = get_dependency_tuple_list_for_recipe(
+                    recipe, blacklist=blacklist
+                )
+            except ValueError:
+                pass
+            adder_first_recipe_name = adding_recipe or name
+
+            # Collect the conflicts:
+            triggered_conflicts = []
+            for dep_tuple_list in deps:
+                # See if the new deps conflict with things added before:
+                if len(set(dep_tuple_list).intersection(
+                       recipe_conflicts)) == len(dep_tuple_list):
```

Comment:
hm why am I feeling checking only the `len` is not exactly what we want to do here?
I'm not 100% sure what you're achieving, don't you want this instead?:
```python
if set(dep_tuple_list) & recipe_conflicts == set(dep_tuple_list):
```
If this is what you wanted to do, not only it's more concise, but also safer. It's just that this `len` thing is confusing me, I can't think of when it would be valid thing to do. But I might be totally off I'm not sure

Reply:
@AndreMiras oops nice catch. yeah your code is what it should be, the `len()` isn't breaking things *in this specific scenario* but useless and error-prone. I'll change it

Reply:
removed it so I hope it's okay if I resolve this one for now :+1:

Reply:
Well now here's another bug, I'm surprise why the test didn't catch it. Can you see the bug introduced? In that same line?